### PR TITLE
[Self-service error codes] Adapt all grpc assertions to additionally take self-service error codes [DPP-680]

### PR DESCRIPTION
**Pull request by tudor-da**
Adapt self-service error codes in conformance tests
 * `Assertions.assertGrpcError` and `assertGrpcErrorRegex` assert self-service error codes based on feature descriptor
 * Adapted all GRPC error assertions to expect self-service error codes as well
 * Added **TODO** and exclusions list to conformance-test-self-service-error-codes in anticipation of some missing self-service error codes implementation. (To be amended in a subsequent PR)

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/Assertions.scala [hunk 1]
```diff
@@ -176,19 +160,23 @@ object Assertions {
     val actualRetryabilitySeconds = actualErrorDetails
       .collectFirst { case err: ErrorDetails.RetryInfoDetail => err.retryDelayInSeconds }
 
+    if (!actualErrorId.contains(expectedErrorId))
```
Comment (mziolekda) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/Assertions.scala:163]: Hmm, this was a problem already before, but what happens when expected error code is a substring of the actual?

Comment (tudor-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/Assertions.scala:163]: That was initially an `Option`.. When changed to String, it still typechecked. Thanks, will change

Comment (mziolekda) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/Assertions.scala]: It makes sense. Thanks

**Review by mziolekda — APPROVED** [review]
Looks good, thanks!

**Pull request merged**